packages/compass-query-bar/src/components/query-bar.tsx

```diff
@@ -0,0 +1,214 @@
+  onReset,
+  queryState,
+  refreshEditorAction,
+  schemaFields,
+  serverVersion,
+  showExportToLanguageButton = true,
+  showQueryHistoryButton = true,
+  toggleExpandQueryOptions,
+  toggleQueryHistory,
+  valid: isQueryValid,
+  ...queryOptionProps
+}) => {
+  const onApply = useCallback(() => {
+    if (isQueryValid) {
+      _onApply();
+    }
+  }, [_onApply, isQueryValid]);
+
+  const onFormSubmit = useCallback(
+    (evt: React.FormEvent) => {
+      evt.preventDefault();
+
+      onApply();
+    },
+    [onApply]
+  );
+
+  return (
+    <form className={queryBarFormStyles} onSubmit={onFormSubmit} noValidate>
+      <div className={queryBarFirstRowStyles}>
+        {showQueryHistoryButton && (
+          <button
+            data-testid="query-history-button"
+            onClick={toggleQueryHistory}
+            className={openQueryHistoryStyles}
+            id="open-query-history"
+            aria-label="Open query history"
+            type="button"
+          >
+            <Icon glyph="Clock" />
+            <Icon glyph="CaretDown" />
+          </button>
+        )}
+        <div className={filterContainerStyles}>
+          <QueryOptionComponent
+            hasError={!queryOptionProps.filterValid}
+            queryOption="filter"
+            onChange={(value: string) => onChangeQueryOption('filter', value)}
+            onApply={onApply}
+            placeholder={
+              queryOptionProps.filterPlaceholder ||
+              OPTION_DEFINITION.filter.placeholder
+            }
+            refreshEditorAction={refreshEditorAction}
+            schemaFields={schemaFields}
+            serverVersion={serverVersion}
+            value={queryOptionProps.filterString}
+          />
+        </div>
+        <Button
+          aria-label="Reset query"
+          data-testid="query-bar-reset-filter-button"
+          onClick={onReset}
+          disabled={queryState !== 'apply'}
+          size="small"
+          type="button"
+        >
+          Reset
+        </Button>
+        <Button
+          data-testid="query-bar-apply-filter-button"
+          disabled={!isQueryValid}
+          variant="primary"
+          size="small"
+          type="submit"
+        >
+          {buttonLabel}
+        </Button>
+        {showExportToLanguageButton && (
+          <IconButton
+            onClick={onOpenExportToLanguage}
+            title="Open export to language"
+            aria-label="Open export to language"
+            data-testid="query-bar-open-export-to-language-button"
+            type="button"
+          >
+            <Icon glyph="Export" />
+          </IconButton>
+        )}
+
+        {queryOptions && queryOptions.length > 0 && (
+          <MoreOptionsToggle
+            aria-controls="additional-query-options-container"
+            data-testid="query-bar-options-toggle"
+            isExpanded={isQueryOptionsExpanded}
+            onToggleOptions={toggleExpandQueryOptions}
+          />
```

Comment:
This toggle is a button, but doesn't have a `type` and so causes a document re-fetch when you toggle it. We probably want it to always be of type button as part of internal component implementation